# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix /fb

# build
make

# install
install -d $DESTDIR/fb/bin
install fbgrab $DESTDIR/fb/bin

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/fbgrab-build.pc << EOF
Name: fbgrab
Version: 1.5
Description: Linux Framebuffer screenshot grabber
Requires: $REQUIRES

exec=\\
/fb/bin/fbgrab
EOF

exit
--- fbgrab-1.5.orig/Configure
+++ fbgrab-1.5/Configure
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      PREFIX=$2
+      shift
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX ?=/s#=.*#= $PREFIX#" Makefile
--- fbgrab-1.5.orig/Makefile
+++ fbgrab-1.5/Makefile
@@ -3,11 +3,13 @@
 ### modular. So this is a simple gnu Makefile...
 ###
 
+PREFIX ?= /usr/local
+
 .DELETE_ON_ERROR:
 .PHONY: install clean all
 
 GZIP := gzip
-GZIPFLAGS := --best --to-stdout
+GZIPFLAGS := -c
 
 all: fbgrab fbgrab.1.gz
 
@@ -18,8 +20,8 @@
 	$(GZIP) $(GZIPFLAGS) $< > $@
 
 install: fbgrab fbgrab.1.gz
-	install -D -m 0755 fbgrab $(DESTDIR)/usr/bin/fbgrab
-	install -D -m 0644 fbgrab.1.gz $(DESTDIR)/usr/share/man/man1/fbgrab.1.gz
+	install -D -m 0755 fbgrab $(DESTDIR)$(PREFIX)/bin/fbgrab
+	install -D -m 0644 fbgrab.1.gz $(DESTDIR)$(PREFIX)/share/man/man1/fbgrab.1.gz
 
 clean:
 	-$(RM) fbgrab fbgrab.1.gz *~ \#*\#
